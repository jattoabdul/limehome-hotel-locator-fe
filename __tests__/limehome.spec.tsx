import * as React from 'react'
import toJson from 'enzyme-to-json'
import {mount, shallow} from 'enzyme'
import LimeHome from '../components/limehome'
import sinon from 'sinon'

//Use array destructurig to create mock functions.
let [onChangeLocationAndMapCenter, setHotels, clearHotels] = new Array(3).fill(jest.fn());

function shallowSetup() {
  // Sample props to pass to our shallow render
  const props = {
    location: '6.522500,3.321350',
    proximity: '1500',
    mapCenter: {lat: 6.522500, lng: 3.321350},
    data: [{name: 'Test Hotel', vicinity: '5 ReactNative Road, Next City', geometry: {location: {lat: 7.522392, lng: 2.123450}}}],
    onChangeLocationAndMapCenter: onChangeLocationAndMapCenter,
    setHotels: setHotels,
    clearHotels: clearHotels
  }
  // wrapper instance around rendered output
  const enzymeWrapper = shallow(<LimeHome {...props} />);

  return {
    props,
    enzymeWrapper
  };
}



describe('Components', () => {
  describe('Shallow rendered LimeHome Component', () => {
    it('should render correctly', () => {
      const { enzymeWrapper, props } = shallowSetup();
      expect(toJson(enzymeWrapper)).toMatchSnapshot();
    });

    it('should render the component with all required elements', () => {
      // Setup wrapper and assign props.
      const { enzymeWrapper, props } = shallowSetup();
      expect(enzymeWrapper.containsMatchingElement(<div className="hambuger" />)).toBe(true);
      expect(enzymeWrapper.find('input')).toHaveLength(2);
      expect(enzymeWrapper.find('input.location-input')).toHaveLength(1);
      expect(enzymeWrapper.find('img.locate-me')).toHaveLength(1);
      expect(enzymeWrapper.find('div.hotel-result-list')).toHaveLength(1);
      expect(enzymeWrapper.find('div.hotel-result-item')).toHaveLength(1);
      expect(enzymeWrapper.find('h2').text()).toBe(props.data[0].name);
      expect(enzymeWrapper.find('.vicinity').text()).toBe(props.data[0].vicinity);
      expect(enzymeWrapper.find('div.map-wrapper')).toHaveLength(1);
    });
  });
})
