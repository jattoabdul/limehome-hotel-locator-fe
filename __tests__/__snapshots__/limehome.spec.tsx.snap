// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Shallow rendered LimeHome Component should render correctly 1`] = `
<Fragment>
  <div
    className="app-wrapper"
  >
    <div
      className="list-wrapper"
    >
      <div
        className="res"
      >
        <div
          className="header-wrapper"
        >
          <img
            alt="logo"
            src="/hotel.svg"
          />
          <div
            className="hambuger"
          />
        </div>
        <form
          onSubmit={[Function]}
        >
          <input
            className="location-input"
            onChange={[Function]}
            type="search"
          />
          <input
            type="submit"
            value="Locate Nearby Hotels"
          />
          <img
            alt="return to location"
            className="locate-me"
            onClick={[Function]}
            src="/target.svg"
            title="Locate me"
          />
        </form>
      </div>
      <div
        className="hotel-result-list"
      >
        <div
          className="hotel-result-item"
          key="0"
        >
          <img
            alt="hotel img"
            src="https://d19yo8val8huli.cloudfront.net/hotels/v7/img/featured/exquisite-suites.jpg"
          />
          <h2>
            Test Hotel
          </h2>
          <p
            className="vicinity"
          >
            5 ReactNative Road, Next City
          </p>
        </div>
      </div>
    </div>
    <div
      className="map-wrapper"
    >
      <SimpleMap
        locations={
          Array [
            Object {
              "geometry": Object {
                "location": Object {
                  "lat": 7.522392,
                  "lng": 2.12345,
                },
              },
              "name": "Test Hotel",
              "vicinity": "5 ReactNative Road, Next City",
            },
          ]
        }
        mapStyles={
          Object {
            "height": "100vh",
            "width": "100%",
          }
        }
        zoomRate={13}
      />
    </div>
  </div>
</Fragment>
`;
